scrunch/datasets.py

```diff
@@ -2386,6 +2386,9 @@ def replace_values(self, variables, filter=None, literal_subvar=False):
                     payload['variables'][self[alias].id] = {'value': val}
         if filter:
             payload['filter'] = process_expr(parse_expr(filter), self.resource)
+            # Remove query parameters from table url
+            table = self.resource.table
+            table.self = table.self[:table.self.find('?')]
```

Comment:
Do we need it only `if filter`?... I know the `limit` query parameter is set when `process_expr` (or another function it depends on) is called, but there's also the possibility that somewhere else, before `replace_values` is called, has also loaded the table, hence the resource's self member will include the `limit` query parameter.

I may be wrong, just consider the possibility and move outside of the `if filter` if you consider it is needed @AlasdairE .

Reply:
Yup, I have moved that outside the `if` now:
https://github.com/Crunch-io/scrunch/pull/385/files#diff-10a14081413b0535e4d0097c2ad71a58R2390

Reply:
LGTM!